packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts

```diff
@@ -874,6 +906,48 @@ export class UserPool extends UserPoolBase {
     class ImportedUserPool extends UserPoolBase {
       public readonly userPoolArn = userPoolArn;
       public readonly userPoolId = userPoolId;
+
+      // In the UserPool construct, the userPoolProviderName is a required property but it is constructed from the arn and id of the user pool.
+      // So we throw an error if it is referenced by a user pool imported using the fromUserPoolId or fromUserPoolArn methods.
+      public get userPoolProviderName(): string {
+        throw new Error('to reference userPoolProviderName, use the `fromUserPoolAttributes`.');
+      }
+
+      constructor() {
+        super(scope, id, {
+          account: arnParts.account,
+          region: arnParts.region,
+        });
+      }
+    }
+
+    return new ImportedUserPool();
+  }
+
+  /**
+   * Import an existing user pool into the stack.
+   */
+  public static fromUserPoolAttributes(scope: Construct, id: string, attrs: UserPoolAttributes): IUserPool {
+    if (!attrs.userPoolArn && !attrs.userPoolId) {
```

Comment:
I'm not sure that this function is really needed here. The fromId and fromArn functions should be able to assemble and return all of the missing attributes. 

Correct me if I'm wrong but the attributes for a UserPool are:
```
Arn: arn:<partition>:cognito-idp:<region>:<account>:userpool/<userpoolId>
ProviderName: cognito-idp.<region>.<url-suffix>/<userpoolId>
ProviderURL: https://cognito-idp.<region>.<url-suffix>/userpoolId/.well-known/jwks.json (this is listed as the token signing key url, which I think is what's being referred to here?)
UserPoolId: userpoolId
```
So, given that, you could add each of the missing fields as attributes in the IUserPool and then construct each of these (assuming same region as in the id for fromId) in the existing fromXxx functions so that you don't even need to create a new function.

Reply:
Doing so would mitigate every one of my comments above.